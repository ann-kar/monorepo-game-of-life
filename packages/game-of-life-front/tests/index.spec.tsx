import { fireEvent, render, getByTestId, getAllByRole} from '@testing-library/react';

import Index from '../pages/index';


describe(Index.name, () => {
    it('should display an empty table after clicking start', () => {
    const context = render(<Index />);
    const buttons = getAllByRole(context.container, "button");
    const board = getByTestId(context.container, "board");
    fireEvent.click(buttons[0]);
    expect(board).toMatchSnapshot();
    })

    it('should display a table with default settings after clicking start (default)', () => {
       const context = render(<Index />);
        const buttons = getAllByRole(context.container, "button");
        const board = getByTestId(context.container, "board");
        fireEvent.click(buttons[1]);
        expect(board).toMatchSnapshot();
        })
});
