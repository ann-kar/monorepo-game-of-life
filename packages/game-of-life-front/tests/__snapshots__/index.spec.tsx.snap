// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index should display a table with default settings after clicking start (default) 1`] = `
<div
  class="board"
  data-testid="board"
>
  <div
    class="row"
  >
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
  </div>
  <div
    class="row"
  >
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell activeCell"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell activeCell"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
  </div>
  <div
    class="row"
  >
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell activeCell"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
  </div>
  <div
    class="row"
  >
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell activeCell"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
  </div>
  <div
    class="row"
  >
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell activeCell"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell activeCell"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
  </div>
  <div
    class="row"
  >
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
  </div>
  <div
    class="row"
  >
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
  </div>
  <div
    class="row"
  >
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
  </div>
  <div
    class="row"
  >
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
  </div>
  <div
    class="row"
  >
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
  </div>
</div>
`;

exports[`Index should display an empty table after clicking start 1`] = `
<div
  class="board"
  data-testid="board"
>
  <div
    class="row"
  >
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
  </div>
  <div
    class="row"
  >
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
  </div>
  <div
    class="row"
  >
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
  </div>
  <div
    class="row"
  >
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
  </div>
  <div
    class="row"
  >
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
  </div>
  <div
    class="row"
  >
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
  </div>
  <div
    class="row"
  >
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
  </div>
  <div
    class="row"
  >
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
  </div>
  <div
    class="row"
  >
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
  </div>
  <div
    class="row"
  >
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
    <div>
      <div
        class="cell false"
        data-testid="cell"
      />
    </div>
  </div>
</div>
`;
